x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtract.java

```diff
@@ -138,16 +162,26 @@ private ChronoField chronoField(FoldContext ctx) {
         return chronoField;
     }
 
-    @Evaluator(warnExceptions = { IllegalArgumentException.class })
-    static long process(long value, BytesRef chronoField, @Fixed ZoneId zone) {
+    @Evaluator(extraName = "Millis", warnExceptions = { IllegalArgumentException.class })
+    static long processMillis(long value, BytesRef chronoField, @Fixed ZoneId zone) {
         return chronoToLong(value, chronoField, zone);
     }
 
-    @Evaluator(extraName = "Constant")
-    static long process(long value, @Fixed ChronoField chronoField, @Fixed ZoneId zone) {
+    @Evaluator(extraName = "ConstantMillis")
+    static long processMillis(long value, @Fixed ChronoField chronoField, @Fixed ZoneId zone) {
         return chronoToLong(value, chronoField, zone);
     }
 
+    @Evaluator(extraName = "Nanos", warnExceptions = { IllegalArgumentException.class })
+    static long processNanos(long value, BytesRef chronoField, @Fixed ZoneId zone) {
+        return chronoToLongNanos(value, chronoField, zone);
```

Comment:
As mentioned before, an alternative to having the util that takes a `BytesRef` would be to convert that here (which would also make it clearer where the `IllegalArgumentException` could stem from) which  and reuse the `processNanos(long, @Fixed ChronoField, @Fixed ZoneId)` below.

Reply:
(Answering both comments)
Just pushed the change, reusing one chronoToLongNanos from the other. I didn't remove the second helper though, as both are really oneliners abstracting some logic (One the `Instant.ofEpochMilli` and the other the `ChronoField.valueOf`). So I feel like if we have one, the other makes sense too.
Also, many converters take BytesRef or Strings in that class, so also to follow the general ""standard"" there